aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,117 @@
+The [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configureservices) method, which is optional, is called by the web host before the `Configure` method to configure the app's services. The web host may configure some services before `Startup` methods are called (details are provided in the [Hosting](xref:fundamentals/hosting) topic). By convention, [Configuration options](xref:fundamentals/configuration/index) are set in the `ConfigureServices` method. Adding services to the service container makes them available within the app and in the `Configure` method resolved via [dependency injection](xref:fundamentals/dependency-injection) or from [IApplicationBuilder.ApplicationServices](/dotnet/api/microsoft.aspnetcore.builder.iapplicationbuilder.applicationservices).
 
-For features that require substantial setup there are `Add[Service]` extension methods on [IServiceCollection](https://docs.microsoft.com/aspnet/core/api/microsoft.extensions.dependencyinjection.iservicecollection). This example from the default web site template configures the app to use services for Entity Framework, Identity, and MVC:
+For features that require substantial setup, there are `Add[Service]` extension methods on [IServiceCollection](/dotnet/api/Microsoft.Extensions.DependencyInjection.IServiceCollection). A typical web app registers services for Entity Framework, Identity, and MVC:
 
 [!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=4,7,11&start=40&end=55)]
 
-Adding services to the services container makes them available within your application via [dependency injection](xref:fundamentals/dependency-injection).
+## Services available in Startup
 
-## Services Available in Startup
+The web host provides some services that are available to the `Startup` class constructor. The app adds additional services via `ConfigureServices`. Both the host and app services are then available in `Configure` and throughout the application.
 
-ASP.NET Core dependency injection provides services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter (but any registered service can be retrieved from this collection, so additional parameters are not necessary).
+Some of the services typically requested by `Startup` methods:
 
-Below are some of the services typically requested by `Startup` methods:
+* In the constructor: `IHostingEnvironment`, `ILogger<Startup>`
+* In `ConfigureServices`: `IServiceCollection`
+* In `Configure`: `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`
```

Comment:
Can you show a snippet with these?

Reply:
Or cut them totally. I don't think this list helps all that much really. The reader is going to see these over and over ... *and over again* in the topics and samples.

Reply:
Cut them.